tcl/pd_bindings.tcl

```diff
@@ -268,13 +271,6 @@ proc ::pd_bindings::patch_bindings {mytoplevel} {
         }
     }
 
-    # <Tab> key to cycle through selection
-    bind $tkcanvas <KeyPress-Tab>        "::pd_bindings::canvas_cycle %W  1 %K %A 0 %k"
-    bind $tkcanvas <Shift-Tab>           "::pd_bindings::canvas_cycle %W -1 %K %A 1 %k"
-    # on X11, <Shift-Tab> is a different key by the name 'ISO_Left_Tab'...
-    # other systems (at least aqua) do not like this name, so we 'catch' any errors
-    catch {bind $tkcanvas <KeyPress-ISO_Left_Tab> "::pd_bindings::canvas_cycle %W -1 %K %A 1 %k" } stderr
-
     # window protocol bindings
```

Comment:
why are they removed?

Reply:
Because with the "goto" functionality you can go directly to any object you want in the patch.

Reply:
so? with your patches you don't need the mouse any more, should we remove that as well ;-) ?
(just to make sure: the answer to this rhetoric question should be "no").

as a sidenote: if you want to remove these bindings (and i obviously don't agree that you should) for de-crufting purposes, you should also remove:
- the `canvas_cycle` tcl-proc
- the `cycleselect` method of the canvas
- the `canvas_cycleselect` function

Reply:
an i find `goto` + trying out all those indices considerably more complicated that just <kbd>Tab</kbd>ing through the patch.

Reply:
The idea with "Go To" is that you don't have to try the indices. You just look at the object, see it's index, type it and hit return.

<kbd>Tab</kbd>-traversal is (pratically) unpredictable because the object order can change. A simple, straight"chain" of objects might have completely "spaced" indices depending on the order you created the objects, the copy-pastes you did, and etc. 

Especially on big patches it can be quite a chore to keep tabbing until you get to the object you want.  Even now that you can see the indices and know how many times you need to press the <kbd>Tab</kbd>. 

What if you must press it 45 times? And holding <kbd>Tab</kbd> is no good either. While holding it did it triggered 30 times? 40 ? 60? I can't see how's that more complicated than

- press <kbd>mod</kbd>+<kbd>g</kbd>
   - all object indices are automatically visible
- look at the object you wanna go and type it's index
-<kbd>Return</kbd>

Also the "Go To" is originally intended for "jumps" since when you're on a "chain" you can just <kbd>mod</kbd>+<kbd>arrow</kbd> your way around with ease.

Reply:
well, this is mostly bike-shedding. but here you go:

> What if you must press it 45 times? And holding Tab is no good either. While holding it did it triggered 30 times? 40 ? 60?

so what if? why would it matter if tab triggered 30 times, 40 or 60? i really don't care as it eventually gets me there.
and *that* is very simple to check with <kbd>Tab</kbd>-cycling. i just keep the key pressed until the object highlights. then i let go. since i was to slow, i back-cycle 3 times. done.
no symbolic processing involved (as in: relate numbers to objects).
to be fair, it might help to have a more contrasting colour that blue/black.

otoh with `goto`, if you have a fairly large patch, having numbers splattered all over the patch won't help readability either: so you have random numbers spilled all over the objects which makes it hard to read both the objects (beneath the numbers) and the numbers themselves. :stuck_out_tongue: